src/Uno.UI/UI/Xaml/UIElementWeakCollection.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Windows.UI.Xaml
+{
+	/// <summary>
+	/// Represents a collection of weak references to UIElement objects.
+	/// </summary>
+	public partial class UIElementWeakCollection : IList<UIElement>, IEnumerable<UIElement>
+	{
+		private readonly List<WeakReference<UIElement>> _innerList = new List<WeakReference<UIElement>>();
```

Comment:
you could use ManagedWeakReference and IWeakReferenceProvider.WeakReference here